tests/Unit/ProjectCodeTest.php

```diff
@@ -83,6 +85,50 @@ public function testProductionClassesHaveUnitTests()
             ]
         );
     }
+
+    public function testControllerActionsUseResponseReturnType()
```

Comment:
I think it may be cleaner to split up this function a bit. Prehaps move the retrieval of controllers to a dataprovider

Reply:
I am not using a data provider here because we want a *single* instead of *n* failure messages.

Reply:
If we want to add more tests, we can extract a method that returns all controllers.